### ISSUE TITLE
ComponentBase 

### ISSUE THREAD

**Issue by BiggA94** [Microsoft/ReSub#100]
short:
Is there any reason, that P in
https://github.com/Microsoft/ReSub/blob/467399dcbbd14fd264b9ef5a0a10f9d3acbf60c1/src/ComponentBase.ts#L44
extends React.Props<any> instead of just {}, as the Component in the Typedefinition of React does? 
See:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cc6de05e3347a9ad3651df02f4b87603758e4398/types/react/index.d.ts#L397

Long form:
I want to user Material-UI together with ReSub.
Material-UI uses augmenting functions for the Properties.
E.g.
``` Typescript
import { WithStyles, createStyles } from '@material-ui/core';

const styles = (theme: Theme) => createStyles({
  root: { /* ... */ },
  paper: { /* ... */ },
  button: { /* ... */ },
});

interface Props extends WithStyles<typeof styles> {
  foo: number;
  bar: boolean;
}
```

So the Component would extend ComponentBase<Props, State>, but this gives a
Compilation error, as in
https://github.com/Microsoft/ReSub/issues/22.
Extending React.props does only work, if on would leave the WithStyles augmentation.
It adds the following:
``` Typescript
interface Props  {
  foo: number;
  bar: boolean;
  classes: {
    foo: string,
    bar: string
  }
}
```
Here, one would loose this convenience method..

The Official Types definition for React.Component (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cc6de05e3347a9ad3651df02f4b87603758e4398/types/react/index.d.ts#L397)
defines the following:
``` Typescript
 interface Component<P = {}, S = {}, SS = any> extends ComponentLifecycle<P, S, SS> { }
```
Is there a particular Reason for having 
``` Typescript
ComponentBase<P extends React.Props<any>, ...
```
instead of just
``` Typescript
ComponentBase<P extends {}, ...
```
in 
https://github.com/Microsoft/ReSub/blob/467399dcbbd14fd264b9ef5a0a10f9d3acbf60c1/src/ComponentBase.ts#L44
?

**Comment by deregtd** [comment]
Initial investigations are that this should be fine -- it appears that those props are now just dealt with in tooling.  Checking with some affected teams within MS to confirm.

**Comment by deregtd** [comment]
We think it's going to be fine.  Thanks for the contribution!

**Comment by BiggA94** [comment]
Nice,
Thank you.

PS: Are version upgrades only triggered manually?

**Comment by deregtd** [comment]
Published 1.1.0
